core/src/main/java/org/apache/iceberg/deletes/ContinuousFileScopedPositionDeleteWriter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.io.DeleteWriteResult;
+import org.apache.iceberg.io.FileWriter;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.CharSequenceSet;
+
+/**
+ * A position delete writer that produces a separate delete file for each referenced data file.
+ *
+ * <p>This writer does not keep track of seen deletes and assumes all incoming records are ordered
```

Comment:
Does it actually assume the records are ordered by file and position? It seems to keep multiple file handles open until the end but assumes records are ordered by position, meaning it is in between clustered and fan-out writers.

Also, why isn't enough to use `ClusteredPositionDeleteWriter` with file granularity?

Reply:
I will have to take a closer look at `BaseTaskWriter`. It has been a while.

Reply:
We do not have a `FileWriterFactory` at hand. What we have is a `FileAppenderFactory` which is already used by the `SortedPosDeleteWriter`.

Also, the internal writer is a `SortedPosDeleteWriter`, which takes care of the sorting for us.

Reply:
Also, I was not brave enough to change any of the existing generic writer classes, as I can not be sure how they are used.
Maybe if we enhance the `SortedPosDeleteWriter` to handle the granularity, then we would not need an extra class. For this decision, I would like to have some oversight from someone who is more familiar with the writer api design and use-cases.